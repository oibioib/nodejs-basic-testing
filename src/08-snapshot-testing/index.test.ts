import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    const linkedListValues = [1, 10, 100, 1000];
    const expectedLinkedList = {
      value: 1,
      next: {
        value: 10,
        next: {
          value: 100,
          next: {
            value: 1000,
            next: {
              value: null,
              next: null,
            },
          },
        },
      },
    };

    const linkedList = generateLinkedList(linkedListValues);
    expect(linkedList).toStrictEqual(expectedLinkedList);
  });

  test('should generate linked list from values 2', () => {
    const linkedListValues = [2, 20, 200, 2000];
    const linkedList = generateLinkedList(linkedListValues);
    expect(linkedList).toMatchSnapshot();
  });
});
