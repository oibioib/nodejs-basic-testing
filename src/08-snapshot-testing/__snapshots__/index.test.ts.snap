// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": null,
          "value": null,
        },
        "value": 2000,
      },
      "value": 200,
    },
    "value": 20,
  },
  "value": 2,
}
`;
